client/src/main/scala/skuber/Volume.scala

```diff
@@ -17,6 +17,16 @@ object Volume {
       directory: Option[String] = None)
     extends Source
 
+  sealed trait ProjectedSource
```

Comment:
There is also a `ServiceAccountTokenProjection` type, although this could be added in a separate PR.

Reply:
added `ServiceAccountTokenProjection`

Reply:
Could we add Json formatting (read/writes) support for the `ServiceAccountTokenProjection` type? Otherwise it will never be written or read.

Reply:
thanks for pointing out... sorry, I'm too spoiled by Circe, there it just wouldn't compile

also added the spec for checking service token mount